klog.go

```diff
@@ -923,13 +935,23 @@ func (l *loggingT) output(s severity, log logr.Logger, buf *buffer, file string,
 		}
 
 		if logging.logFile != "" {
+			sb, ok := logging.file[s].(*syncBuffer)
 			// Since we are using a single log file, all of the items in l.file array
 			// will point to the same file, so just use one of them to write data.
 			if l.file[infoLog] == nil {
 				if err := l.createFiles(infoLog); err != nil {
 					os.Stderr.Write(data) // Make sure the message appears somewhere.
 					l.exit(err)
 				}
+			} else if ok {
+				// Determine whether the file has been deleted or moved
+				// If the file does not exist, rebuild it
+				if !exists(logging.logFile) {
+					if err := sb.rotateFile(time.Now(), true); err != nil {
+						os.Stderr.Write(data) // Make sure the message appears somewhere.
+						l.exit(err)
+					}
+				}
```

Comment:
I entered your code in version 19 and found that there is no way to solve this problem. The reason is that in the process of rebuilding the handle, because sb.file has a value, it will ran a rename method.
At this time, because the file does not exist, errors will continue to be reported.
func (sb *syncBuffer) rotateFile(now time.Time, startup bool) error {
	if sb.file != nil {
		io.WriteString(os.Stdout, fmt.Sprint("========rotateFile -> flush========\n"))
		sb.Flush()
		sb.file.Close()
		if err := sb.renameLogFile(); err != nil {
			return err
		}
	}
I added the emptying operation.

sb.file=nil
if err := sb.rotateFile(time.Now(), true); err != nil {

Purpose achieved

Maybe I understand something wrong, maybe not, please advise!

Reply:
That was really strange, I tested it locally, it was OK, and also added the test code, the test was successful

Reply:
Let me test it again

Reply:
Where is the `renameLogFile` function located ？I didn't find it in the master branch.
Can you give me the link address of the above code ？

Reply:
sorry，my teamates add it. it will be useful to add your code after a new test. Thanks ! ! !